lib/matplotlib/axes/_base.py

```diff
@@ -2996,21 +3000,24 @@ def _update_title_position(self, renderer):
             locator = ax.get_axes_locator()
             ax.apply_aspect(locator(self, renderer) if locator else None)
 
+        top = -np.inf
+        for ax in axs:
+            bb = None
+            xticklabel_top = any(tick.label2.get_visible() for tick in
+                                 [ax.xaxis.majorTicks[0], ax.xaxis.minorTicks[0]])
```

Comment:
Directly check the visibility of the tick label as I think that is the most relevant part from `xaxis.get_ticks_position()` and will prevent the unnecessary calls to `xaxis.get_tightbbox` mentioned at https://github.com/matplotlib/matplotlib/issues/26150#issuecomment-2132256785.  Only checking the first tick is consistent with [get_ticks_position](https://github.com/matplotlib/matplotlib/blob/b02aa82f3d3ba7f55e8a3fd431bc8763e3ae5c5c/lib/matplotlib/axis.py#L2250-L2251).

There is a question in my mind about what happens if the ticks are pointing outward but unlabelled, but I _think_ that could already be an issue with the existing approach as you could have "default" position and outward ticks.